tensorrt_llm/executor/base_worker.py

```
@@ -715,20 +715,112 @@ def _check_sleep_wakeup_preconditions(self, method: str) -> None:
+           finished their local VMM operations.
+        5. Exit ``control_action()``, resuming rank-0's event loop.
+
+        Args:
+            action: ``"sleep"`` or ``"wakeup"``.
+            tags: Parsed :class:`~tensorrt_llm.llmapi.llm_args.ExecutorMemoryType`
+                values; forwarded verbatim to each rank's VMM call.
+        """
+        from tensorrt_llm._torch.pyexecutor.py_executor import (
+            _CONTROL_ACK_TAG, _CONTROL_ACTION_TAG)
+        from tensorrt_llm._torch.virtual_memory import (materialize_with_tag,
+                                                         release_with_tag)
```

Flake8 (7.3.0)

[error] continuation line over-indented for visual indent

(E127)